ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.ledger.api.auth
+
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+
+import spray.json._
+
+/** The JWT token payload used in [[AuthServiceJWT]]
+  *
+  *
+  * @param ledgerId       If set, the token is only valid for the given ledger ID.
+  *                       May also be used to fill in missing ledger ID fields in ledger API requests.
+  *
+  * @param participantId  If set, the token is only valid for the given participant ID.
+  *                       May also be used to fill in missing participant ID fields in ledger API requests.
+  *
+  * @param applicationId  If set, the token is only valid for the given application ID.
+  *                       May also be used to fill in missing application ID fields in ledger API requests.
+  *
+  * @param exp            If set, the token is only valid before the given instant.
+  *                       Note: This is a registered claim in JWT
+  *
+  * @param admin          Whether the token bearer is authorized to use admin endpoints of the ledger API.
+  *
+  * @param actAs          List of parties the token bearer can act as.
+  *                       May also be used to fill in missing party fields in ledger API requests (e.g., submitter).
+  *
+  * @param readAs         List of parties the token bearer can read data for.
+  *                       May also be used to fill in missing party fields in ledger API requests (e.g., transaction filter).
+  */
+case class AuthServiceJWTPayload(
+    ledgerId: Option[String],
+    participantId: Option[String],
+    applicationId: Option[String],
+    exp: Option[Instant],
+    admin: Boolean,
+    actAs: List[String],
```

Comment:
@stefanobaghino-da: We can probably restrict this to`Optional[String]` because we anyway don't support commands with multiple submitters. WDYT?

Reply:
We don't support filling in the information for the user but you can still have multiple parties, right? And you would have to have them on the token _and_ provide them in the API call, if I'm not mistaken.

Reply:
👍